next-cloudinary/src/components/CldUploadWidget/CldUploadWidget.tsx

```diff
@@ -125,6 +125,15 @@ const CldUploadWidget = ({
     });
   }
 
+  useEffect(() => {
+    handleOnLoad();
```

Comment:
@JoshuaRotimi we actually don't need this here - Next.js works a little bit differently than vanilla React in that it has the Script component, so this function gets invoked as soon as the script is ready that handles initialization

that means that this function could actually possibly create an error (might be happening on the other work you're doing) in that it could potentially try to initialize without the Script being ready

Reply:
Oh alright. After trying multiple things, I found out that calling the `handleOnLoad` function in the useEffect fixes the rendering issue when you navigate between pages with the `CldVideoPlayer` component. I would further check to see if that is what causes the error on the examples repo that you mentioned.

Reply:
i tried commenting out that line and it seemed to work without it, just with the <Script tag natively reloading between the pages - where are you seeing an issue?

Reply:
Oh sorry. Not on this one. The comment should have been on the other component `CldVideoPlayer` component not the `CldUploadWidget`. You are correct. It should work without that on the `CldUploadWidget`.

Reply:
got it okay - i would expect the same for the video player though, ill test locally too